parser.go

```diff
@@ -135,7 +143,12 @@ func (p *Parser) ParseUnverified(tokenString string, claims Claims) (token *Toke
 		}
 		return token, parts, newError("could not base64 decode header", ErrTokenMalformed, err)
 	}
-	if err = json.Unmarshal(headerBytes, &token.Header); err != nil {
+	if p.jsonEncoder != nil {
+		err = p.jsonEncoder.Unmarshal(headerBytes, &token.Header)
```

Comment:
I think it would be cleaner to set `p.jsonEncoder` to `json.Unmarshal` (if nil) and then call `p.jsonEncoder.Unmarshal()` at the end like

```go
if p.jsonEncoder == nil {
  p.jsonEncoder = json.Unmarshal
}

err = p.jsonEncoder.Unmarshal(headerBytes, &token.Header)
```